doc/nrf/ug_radio_fem.rst

```diff
@@ -62,12 +62,80 @@ Before you add the devicetree node in your application, complete the following s
+   The feature is made available in its current state, but the design and interfaces can change between release tags.
+   The feature is also labeled as ``EXPERIMENTAL`` in Kconfig files to indicate this status.
+
+To use this model, set :kconfig:option:`CONFIG_MPSL_FEM_POWER_MODEL` and :kconfig:option:`CONFIG_MPSL_FEM_POWER_MODEL_NRF21540_USE_BUILTIN` to ``y``.
+
+This feature uses a model to compensate the FEM gain for the following external conditions:
+
+* Temperature
+* FEM supply voltage
+* Carrier frequency
+* FEM input power.
+
+The model assumes that the FEM supply voltage is constant.
+To provide the value of this voltage to the MPSL subsystem, use the :kconfig:option:`CONFIG_MPSL_FEM_POWER_VOLTAGE` option.
+
+Adding custom power models
+^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+If the way MPSL splits the TX power into components does not meet your requirements, or if you wish to implement a custom compensation model, you can provide one as follows:
+
+1. Set :kconfig:option:`CONFIG_MPSL_FEM_POWER_MODEL` to ``y``
+#. Provide an implementation of the ``mpsl_fem_power_model_to_use_get()`` function.
+   This function should return a pointer to a variable of the type ``mpsl_fem_power_model_t`` which contains pointers to the model's callbacks.
+#. Mandatorily implement the model's ``fetch`` callback (details explained below).
+#. Optionally implement the model's ``init`` callback (details explained below).
+   If no ``init`` callback is provided, pass ``NULL`` as the pointer to the callback.
+#. You can also optionally extend the ``MPSL_FEM_POWER_MODEL_CHOICE`` Kconfig choice with an option to select your custom model, for example, if you want to test multiple custom models.
+
+The ``init`` callback is called by MPSL once, after FEM configuration finishes.
+Calibration data (acquired from FEM internal registers, Kconfig options, and devicetree files) is passed to this function using a parameter of the ``mpsl_fem_calibration_data_t`` type.
+The meaning of the calibration data stored in this parameter is implementation-specific.
+For details, see the ``mpsl_fem_calibration_data_t`` type documentation.
+
+The ``fetch`` callback is used to split the power between the SoC output power and the FEM gain.
+It is called every time this split needs to be recalculated.
+For 802.15.4, this happens before every transmission.
+For Bluetooth® Low Energy, this happens every time the channel changes.
+
+.. note::
+   This function is called in a time-critical path.
+   Please refer to the documentation of ``mpsl_fem_power_model_t`` on timing constraints.
+   Any complex calculations have to be done outside this function (for example, using a look up table).
+   Failing to meet the timing requirements will lead to an undefined behavior of the protocol stacks.
+
+
+The ``fetch`` callback must fill out all the fields of the  the ``p_output`` output parameter.
+For more details, see the ``mpsl_fem_power_model_output_t`` type documentation.
```

Comment:
Where is this "mpsl_fem_power_model_output_t" documentation located?

Reply:
As answered in https://github.com/nrfconnect/sdk-nrf/pull/8536#discussion_r957477706